public/app/features/datasources/state/actions.ts

```diff
@@ -13,14 +13,84 @@ import {
   dataSourcePluginsLoad,
   dataSourcePluginsLoaded,
   dataSourcesLoaded,
+  initDataSourceSettingsFailed,
+  initDataSourceSettingsSucceeded,
+  testDataSourceStarting,
+  testDataSourceSucceeded,
+  testDataSourceFailed,
 } from './reducers';
 import { buildCategories } from './buildCategories';
+import { getDataSource, getDataSourceMeta } from './selectors';
 
 export interface DataSourceTypesLoadedPayload {
   plugins: DataSourcePluginMeta[];
   categories: DataSourcePluginCategory[];
 }
 
+export const initDataSourceSettings = (pageId: number): ThunkResult<void> => {
+  return async (dispatch: ThunkDispatch, getState) => {
+    if (isNaN(pageId)) {
+      // this.setState({ loadError: 'Invalid ID' });
+      dispatch(initDataSourceSettingsFailed('Invalid ID'));
+      return;
+    }
+
+    try {
+      await dispatch(loadDataSource(pageId));
+      if (getState().dataSourceSettings.plugin) {
+        return;
+      }
+
+      const dataSource = getDataSource(getState().dataSources, pageId);
+      const dataSourceMeta = getDataSourceMeta(getState().dataSources, dataSource.type);
+      const importedPlugin = await importDataSourcePlugin(dataSourceMeta);
```

Comment:
I think we talked about this yesterday, the original code had an try...catch that just console.loggged out a message and I was not sure why. I think looks a lot cleaner but just want to make sure we're not introducing an unexpected issue with this.
```typescript
try {	
      await this.importPlugin(makePromiseCancelable(this.importDataSourcePlugin(dataSourceMeta)));	
    } catch (e) {	
      console.log('Failed to import plugin module', e);	
    }
```